airflow/providers/google/cloud/operators/kubernetes_engine.py

```diff
@@ -335,8 +339,16 @@ def __init__(
                 stacklevel=2,
             )
             is_delete_operator_pod = False
+        if in_cluster:
+            warnings.warn(
+                f"You should not set parameter `in_cluster` `True` in class {self.__class__.__name__}. "
+                "This operator uses Google Service Account(GSA) not Kubernetes Service Account(KSA)."
+                "If you can use KSA with in_cluster `True`, you should use KubernetesPodOperator directly.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
this can be explained in the operator documentation this is not really a deprecation warning.